bundles/org.openhab.binding.pulseaudio/src/main/java/org/openhab/binding/pulseaudio/internal/PulseAudioAudioSource.java

```diff
@@ -133,24 +135,46 @@ private synchronized void registerPipe(PipedOutputStream pipeOutput) {
         startPipeWrite();
     }
 
-    private void startPipeWrite() {
-        if (pipeWriteTask == null) {
+    private synchronized void startPipeWrite() {
+        if (this.pipeWriteTask == null) {
             this.pipeWriteTask = executor.submit(() -> {
                 int lengthRead;
                 byte[] buffer = new byte[1024];
+                int readRetries = 3;
                 while (!pipeOutputs.isEmpty()) {
                     var stream = getSourceInputStream();
                     if (stream != null) {
                         try {
                             lengthRead = stream.read(buffer);
+                            readRetries = 3;
```

Comment:
Is a retry mechanism really required? 
During my previous tests  I never entered in that case. Did you? 
If required, is it a workaround for a bad implementation?

Reply:
I realize that after a system sleep on the openHAB server the socket doesn't report to be closed but you can't read more from it (it gives a read timeout exception on all the read calls). I think there is some cases where the socket just hangs and became unusable but the jvm is not aware of it.
I saw a similar retry mechanism somewhere else in this binding where the socket disconnection is forced after 3 consecutive IOException so I made the same here and it solves the problem.

> If required, is it a workaround for a bad implementation?

So I think the answer is no, or at least not a bad implementation inside the binding.